doc/conf.py

```diff
@@ -179,12 +184,28 @@
 # further.  For a list of options available for each theme, see the
 # documentation.
 html_theme_options = {
+    "dark_css_variables": {
+        "color-announcement-background": "#935610",
+        "color-announcement-text": "#FFFFFF",
+    },
     "light_css_variables": {
         "admonition-font-size": "100%",
-        "admonition-title-font-size": "100%"
-    }
+        "admonition-title-font-size": "100%",
+        "color-announcement-background": "#FBB360",
+        "color-announcement-text": "#111418",
+    },
 }
 
+# Add banner in case version is not stable
+if "dev" in release:
+    html_theme_options["announcement"] = (
+        f'<p>This is the development documentation of nilearn ({release}) '
+        '<a class="sd-sphinx-override sd-badge sd-text-wrap '
+        'sd-btn-outline-dark reference external" '
+        'href="https://nilearn.github.io">'
+        f'<span>Switch to stable version ({version})</span></a></p>'
+    )
```

Comment:
**A note just so we don't forget (can be done in another PR)**

Since it might be annoying to users to not be able to switch back to dev from stable we can either:

- Add an `else` statement to define the same thing but inverted for the stable docs
- You also mentioned the possibility of a drop down menu to pick from all versions (would require building previous doc versions and deploying to nilearn.github.io)

In any case, changes to stable wouldn't show up until the next release so we have time to think about it.

Reply:
Also, users could expect this button to take them to the equivalent page they're currently looking at in the stable docs. In this PR, the button simply takes you to the homepage of the stable docs.

Reply:
Also, just as a note, maybe this repo could help us deal with multiple versions: https://github.com/Holzhaus/sphinx-multiversion

Reply:
> You also mentioned the possibility of a drop down menu to pick from all versions (would require building previous doc versions and deploying to nilearn.github.io)

Personally I'd lean against this, since I think it will require storing a large number of (largely duplicate) artifacts. But if others feel strongly on this, it'd be nice to unpack the benefits a bit more !